oeps/oep-0017-feature-toggles.rst

```diff
@@ -0,0 +1,927 @@
+users. For example, the state of a feature toggle may depend on whether the user
+is a paying customer or applicable to a white label site.
+
+Use Case 7: Opt-out
+^^^^^^^^^^^^^^^^^^^
+.. _Opt-out: `Use Case 7: Opt-out`_
+
+In an extreme case, the business may choose to keep a feature disabled for a certain
+group (e.g., for a course or for an organization) in order to appease concerns about
+the change. However, as this introduces a roadblock to removing a toggle and its 
+corresponding complexity, further effort should be made to tweak the feature to
+accommodate the group's concerns and/or to make the group more comfortable with the 
+change.
+
+Use Case 8: Open edX option
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+.. _Open edX option: `Use Case 8: Open edX option`_
+
+When a team implements a feature that they do not expect to be adopted by all Open
+edX instances, they may introduce a toggle to gate the feature. However, since there
+is a large cost to supporting long-term toggles, the following alternatives should
+be considered:
+
+* A management command to convert an old mechanism to a new one.
+
+* Keep the toggle around for only 1 additional Open edX named release, providing
+  Open edX operators the ability to rollout the change on their own systems.
+
+* A design pattern such as a plug-in architecture that does not require code 
+  deployment toggles. 
+  
+**Note:** Remember that feature toggles are not a substitute for clean architecture
+and SOLID design principles. Any long-term feature toggle should be carefully
+considered along with architectural patterns such as plugins, dependency injections,
+separable services and libraries with clear interfaces. Sometimes the need for a
+toggle can be completely eliminated. Other times the toggle may still need to exist
+but with much less complexity.
+
+(Out of Scope) Use Case: Experimentation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Note that we are excluding experiment toggles from this list of use cases.
+Experiment toggles are used to perform multivariate (A/B) testing in order
+to generate statistically significant results to make data-driven optimizations and
+feature changes. Users are placed in different experimentation groups that are 
+associated with different code paths. The effectiveness of each code path is then
+evaluated by measuring its impact on users' aggregate behavior.
+
+This is a deeper topic that is worth exploring in a separate OEP (see `Optimizely
+Tips and Tricks`_). For now, suffice it to say that edX uses an external A/B testing 
+platform (Optimizely_) to serve this purpose. Among other things, Optimizely supports
+user segmentation and targeting, data aggregation capabilities, statistical tools,
+and toggled code customizations. At this time, Optimizely is used for edX 
+experimentation, customizing edX code, but without merging any changes to the edX
+codebase. This also contrasts with the use cases that are in scope of this OEP. 
+
```

Comment:
I'd embed some sort of clarification like the following:
```
*Note:* An experiment toggle is a very specific toggle type. This section does not imply that the other uses cases may still come in handy while working on an experiment.
```

Reply:
done